features/appmgt/org.wso2.carbon.appmgt.store.feature/src/main/resources/store/extensions/assets/webapp/pages/assets.jag

```diff
@@ -217,23 +217,27 @@ require('/modules/store.js').exec(function (ctx) {
                     var appMgtDaoService = Packages.org.wso2.carbon.appmgt.impl.dao.AppMDAO;
                     var appMgtDaoObj = new appMgtDaoService();
 
-                    var result = appMgtDaoObj.getAppsByHitCount(userName, (paging.start + index), 12);
+                    var result = appMgtDaoObj.getAppsByHitCount(userName, (paging.start + index), PAGE_SIZE);
                     if (log.isDebugEnabled()) {
-                        log.debug("Query executed successfully to get the assets hit count");
+                        var recordCount = 0;
+                        if (result != null) {
+                            recordCount = result.length;
+                        }
+                        log.debug("Query executed successfully to get the assets hit count. " + recordCount
+                        + " records found");
                     }
                     index += 12;
                     return JSON.parse(result);
                 };
 
                 assets = [];
-                var aid, asset;
-                while (assets.length < 12 && maxTry < 10) {
+                while (assets.length < PAGE_SIZE && maxTry < 10) {
                     maxTry++;
                     var result = getNextassetSet();
-                    for (var n = 0; n < result.length && assets.length < 12; n++) {
-                        aid = String(result[n].UUID);
+                    for (var n = 0; n < result.length && assets.length < PAGE_SIZE; n++) {
```

Comment:
No need to check whether result is null every time. Do it prior to the for loop.